traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -688,7 +686,7 @@ func (cg *TOCacheGroup) handleCoordinateUpdate() (*int, error, error, int) {
 	}
 
 	if err = cg.updateCoordinate(); err != nil {
-		return nil, nil, tc.DBError, http.StatusInternalServerError
+		return nil, err, tc.DBError, http.StatusBadRequest
```

Comment:
So since this might actually be a database error, this should still return an Internal Server Error in my opinion. 
What you could do is add a check at the start of the `Update` method checking to see if the cachegroup with the given name(from the request body) has the same ID as that passed in the request. If true, then proceed, else, return a 400 or 409 status code with the correct error (something like `cachegroup name should be unique`).

Also, could you pls add some tests for this?

Reply:
Additionally, the `tc.DBError` is now deprecated. You can just return `err` in its place, and `nil` in place of the user error.